subsys/bluetooth/controller/hci/hci.c

```diff
@@ -2590,10 +2590,12 @@ static void le_df_set_cl_cte_enable(struct net_buf *buf, struct net_buf **evt)
 static void le_df_set_cl_iq_sampling_enable(struct net_buf *buf, struct net_buf **evt)
 {
 	struct bt_hci_cp_le_set_cl_cte_sampling_enable *cmd = (void *)buf->data;
-	uint16_t sync_handle;
+	struct bt_hci_rp_le_set_cl_cte_sampling_enable *rp;
+	uint16_t sync_handle, sync_handle_le16;
 	uint8_t status;
 
-	sync_handle = sys_le16_to_cpu(cmd->sync_handle);
+	sync_handle_le16 = cmd->sync_handle;
```

Comment:
I'd skip this optimization and just convert it back again when you set it.
On ARM devices this is a no-op anyway.

Reply:
You are right, assembly generated is the same with or without additional variable. Changes as suggested.